warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java

```diff
@@ -387,6 +397,169 @@ public void addFilters(AttributeFactory attributeFactory, Map<String,EventDataQu
             return new Set[] {fields, terms};
         }
         
+        /**
+         * Distributes a function node into one or more components of an index query. If applicable, the function node will be updated with the field of the
+         * intersected component of index query.
+         * <p>
+         * For example, for a function that expands into field F1
+         * 
+         * <pre>
+         * content:phrase(termOffsetMap, 'foo', 'bar')
+         * </pre>
+         * 
+         * becomes
+         * 
+         * <pre>
+         * (content:phrase(F1, termOffsetMap, 'foo', 'bar') &amp;&amp; F1 == 'foo' &amp;&amp; F2 == 'bar'))
+         * </pre>
+         *
+         * @param function
+         *            a function of indeterminate namespace
+         * @param indexQuery
+         *            an intersection of equality nodes or a union of such nodes
+         * @return the root of a distributed index query
+         */
+        public static JexlNode distributeFunctionIntoIndexQuery(JexlNode function, JexlNode indexQuery) {
+            
+            List<JexlNode> components = new LinkedList<>();
+            
+            JexlNode deref = JexlASTHelper.dereference(indexQuery);
+            if (deref instanceof ASTAndNode) {
+                // single component
+                components.add(deref);
+            } else if (deref instanceof ASTOrNode) {
+                // multiple components
+                for (JexlNode child : JexlNodes.children(deref)) {
+                    components.add(JexlASTHelper.dereference(child));
+                }
+            } else {
+                throw new IllegalStateException("Expected a dereferenced IndexQuery node to be an AND or OR node, but was " + deref.getClass().getSimpleName());
```

Comment:
I fell that this shoule be "to be an ASTAndNode or ASTOrNode" -- since the classname is what will print.